enterprise/internal/batches/reconciler/reconciler.go

```diff
@@ -66,6 +66,7 @@ func (r *Reconciler) HandlerFunc() workerutil.HandlerFunc[*btypes.Changeset] {
 // errored and set its FailureMessage to the error.
 func (r *Reconciler) process(ctx context.Context, logger log.Logger, tx *store.Store, ch *btypes.Changeset) error {
 	// Reset the error message.
+	ch.PreviousFailureMessage = ch.FailureMessage
 	ch.FailureMessage = nil
```

Comment:
I actually wonder if we should wrap this in a conditional like:

```suggestion
	if ch.FailureMessage != nil {
		ch.PreviousFailureMessage = ch.FailureMessage
		ch.FailureMessage = nil
	}
```

My reason for suggesting this is that I _think_ since `store.EnqueueChangeset()` also copies it over and sets `FailureMessage` to `nil`, we might be doing this twice for changesets enqueued that way, so this might happen:

1. We call `store.EnqueueChangeset(ctx, changeset_1, ...)` for a changeset that had an error
    - `changeset_1.PreviousFailureMessage` gets set to the error
    - `changeset_1.FailureMessage` gets set to `nil`
2. Reconciler worker picks up `changeset_1` and calls `process(ctx, logger, tx, changeset_1)` here
    - `changeset_1.FailureMessage` is already `nil`, so `changeset_1.PreviousFailureMessage` gets set to `nil`
    - (not that it matters) `changeset_1.FailureMessage` gets set to `nil` again

And thus we lose the `PreviousFailureMessage`.

With a check like this first, we could ensure we're not unintentionally clearing out the `PreviousFailureMessage` if we already did that whole copy + reset dance.